checkstyle-checks/src/main/resources/sonatype/checkstyle-configuration.xml

```diff
@@ -0,0 +1,181 @@
+<!--
+
+    Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
+    Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
+    "Sonatype" is a trademark of Sonatype, Inc.
+
+-->
+<!DOCTYPE module PUBLIC
+    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+<module name="Checker">
+  <property name="charset" value="UTF-8"/>
+
+  <property name="severity" value="error"/>
+
+  <property name="fileExtensions" value="java"/>
+  <module name="FileTabCharacter">
+    <property name="eachLine" value="true"/>
+  </module>
+
+  <module name="NewlineAtEndOfFile"/>
+
+  <module name="SuppressWarningsFilter" />
+
+  <module name="TreeWalker">
+    <module name="com.sonatype.checks.ClassStructureEmptyLineCheck" />
+    <module name="SuppressWarningsHolder" />
+    <module name="UnnecessaryParentheses"/>
```

Comment:
I trust this only fires on the simplest expressions and doesn't prevent us from using semantically unnecessary parentheses where we feel it helps comprehension of the expression evaluation.

Reply:
> I trust this only fires on the simplest expressions and doesn't prevent us from using semantically unnecessary parentheses where we feel it helps comprehension of the expression evaluation.

I believe these javadocs describe the cases that it detects: http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.html

Reply:
> I trust this only fires on the simplest expressions and doesn't prevent us from using semantically unnecessary parentheses where we feel it helps comprehension of the expression evaluation.

_From the docs:_

Checks if unnecessary parentheses are used in a statement or expression. The check will flag the following with warnings:

     return (x);          // parens around identifier
     return (x + 1);      // parens around return value
     int x = (y / 2 + 1); // parens around assignment rhs
     for (int i = (0); i < 10; i++) {  // parens around literal
     t -= (z + 1);        // parens around assignment rhs</pre>

The check is not "type aware", that is to say, it can't tell if parentheses are unnecessary based on the types in an expression.  It also doesn't know about operator precedence and associativity; therefore it won't catch something like

     int x = (a + b) + c;</pre>

In the above case, given that a, b, and c are all int variables, the parentheses around a + b are not needed.